File: java/src/poop_project/Pixel.java
package poop_project;

public class Pixel {
	
	int red = 0;
	int green = 0;
	int blue = 0;
	int alpha = 255;
	
	public Pixel(int r, int g, int b, int a) {
		super();
		red = r;
		green = g;
		blue = b;
		alpha = a;
	}
	
	public Pixel() {}

	public int getAlpha() {
		return alpha;
	}

	public int getRed() {
		return red;
	}

	public int getGreen() {
		return green;
	}

	public int getBlue() {
		return blue;
	}
		
}

File: cpp/Pixel.cpp
#include"Pixel.h"
#include <algorithm>

Pixel::Pixel() {}
Pixel::Pixel(int r, int g, int b, int a) :red(r), green(g), blue(b), alpha(a) {}

int Pixel::getRed() const { return red; }
int Pixel::getBlue() const { return blue; }
int Pixel::getGreen() const { return green; }
int Pixel::getAlpha() const { return alpha; }

void Pixel::round() { //do i need this?
	red = std::min(255, std::max(0, red));
	green = std::min(255, std::max(0, green));
	blue = std::min(255, std::max(0, blue));
}